addons/html_editor/controllers/main.py

```diff
@@ -362,7 +362,9 @@ def modify_image(self, attachment, res_model=None, res_id=None, name=None, data=
         inserted into the DOM.
         """
         self._clean_context()
-        attachment = request.env['ir.attachment'].browse(attachment.id)
+        attachment = request.env['ir.attachment'].search([('id', '=', attachment.id), ('name', '!=', False)], limit=1)
+        if not attachment:
+            return "/html_editor/static/src/img/placeholder_thumbnail.png?placeholder=1"
```

Comment:
In stable, let's add an optional parameter to the method to request the new behavior. If unset: fail like before.
And in the call from JS, add that parameter to request this new behavior.
Also add comments about removing those in master.

Reply:
@paru-odoo I guess you can put such a parameter in the context - this way it won't risk being lost by overrides.